lilac2/api.py

```diff
@@ -58,8 +58,9 @@ def add_into_array(which: str, extra_deps: Iterable[str]) -> None:
   '''
   field_appeared = False
 
+  pattern = re.compile(which + r'=')
```

Comment:
need to use `re.escape`.

Reply:
Can't we assume the `which` is a valid shell variable name so it won't contain regex metacharacters?

Reply:
It's better to escape. Maybe someone will try to modify `depends+=(...)` lines?